src/main/java/org/motechproject/mots/service/SchedulerService.java

```diff
@@ -0,0 +1,67 @@
+package org.motechproject.mots.service;
+
+import org.motechproject.mots.dto.AutomatedReportSettingsDto;
+import org.motechproject.mots.exception.AutomatedReportException;
+import org.motechproject.mots.scheduler.BaseJob;
+import org.motechproject.mots.scheduler.SendReportJob;
+import org.quartz.JobDetail;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.Trigger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.quartz.SchedulerFactoryBean;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SchedulerService {
+
+  @Autowired
+  private SchedulerFactoryBean schedulerFactoryBean;
+
+  @Autowired
+  private SendReportJob sendReportJob;
+
+  public void updateJob(AutomatedReportSettingsDto reportSettingsDto) {
```

Comment:
when you call this in the automated report service instead of controller, you should check if the start date or interval was changed and only then reschedule the job